### ISSUE TITLE
time and memory cost more and more while using keras.backend.ctc_decoder()

### ISSUE THREAD

**Issue by jkmiao** [tensorflow/tensorflow#10218]
### memory not released bug with tf.python.ops.gen_ctc_ops  both with ctc_greedy_decoder and ctc_beam_search_decoder...

```python
import time

from keras import backend as K

start_time = time.time()
while True:
     y_pred  = K.ctc_decode(args)
     cost_time = time.time()-start_time
     print 'cost time', cost_time   # time cost more and more with the same input args. why ??!! 
 ...
```
when using  K.ctc_decode() in a large loop, time cost longer  loop become larger...

- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: ubuntu 14.04
- **TensorFlow installed from (source or binary)**: source ( pip install -U tensorflow)
- **TensorFlow version (use command below)**: (1.1.0)
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: None ( same problem occured if has CUDA7.5)
- **GPU model and memory**: None ( same problem occured with GPU of 32G)
- **Exact command to reproduce**:

####  function in file  tensorflow.python.ops.ctc_ops.py

Please help, thank you very much!

**Comment by aselle** [comment]
Could you provide the full code?

**Comment by cccong** [comment]
@aselle I have the same problem.  Here is my code.
`import string
from io import BytesIO
import base64

import numpy as np
from keras.models import load_model
from keras import backend as K
from scipy import misc


characters = string.digits + string.ascii_lowercase + ' '
width, height, n_len = 170, 60, 8
base_model = load_model('model.h5')

ctc_g = []


def predict(img_base64=None):
    pfd = BytesIO(base64.b64decode(img_base64))
    img = misc.imread(pfd, mode='RGB')
    pfd.close()
    img = misc.imresize(img, (height, width))
    img = img.transpose(1, 0, 2)
    img = img.reshape(1, width, height, 3)
    y_pred = base_model.predict(img)
    y_pred = y_pred[:, 2:, :]

    # out = np.argmax(y_pred[0], axis=1)
    # out = ''.join([characters[x] for x in out if x!=36])

    ctc_code = K.ctc_decode(y_pred, input_length=np.ones(y_pred.shape[0]) * y_pred.shape[1])
    out = K.get_value(ctc_code[0][0])
    out = out[:, :n_len]
    out = np.array(out, dtype=np.int8)
    out = ''.join([characters[x] for x in out[0]])

    return out
`
After the function `predict` called abouts hundreds of times, it slows down, momery just keep going up.
